### ISSUE TITLE
Run Boundary test on gcode with relative coordinates isn't working right.

### ISSUE THREAD

**Issue by NomadicVoxel** [Snapmaker/Snapmaker2-Controller#117]
... At least, it isn't when equipped with lasers. 

This behavior happens consistently with code that involves G91, a code that labels following coordinates as relative to the tool rather than relative to the origin. 
The output when actually running the code works as expected, it seems to be specifically the "run boundary" test. 

Note my [conversation on the forum](https://forum.snapmaker.com/t/anyone-know-how-to-get-lightburn-to-use-work-origin-the-way-luban-does/18868/5) for particulars for my case. Settings in LightBurn were to have the project centered around work origin, which does yield similar behavior to Luban for printing. It's specifically relative coordinates in the Run Boundary test.

**Comment by brent113** [comment]
Agreed. I previously mentioned this issue here: https://github.com/Snapmaker/Snapmaker2-Controller/issues/25#issuecomment-743694165
